platform/cog-platform-drm.c

```diff
@@ -236,6 +279,180 @@ init_egl ()
 }
 
 
+static void
+input_handle_key_event (struct libinput_event_keyboard *key_event)
+{
+    struct wpe_input_xkb_context *default_context = wpe_input_xkb_context_get_default ();
+    struct xkb_state *state = wpe_input_xkb_context_get_state (default_context);
+
+    uint32_t key = libinput_event_keyboard_get_key (key_event) + 8;
+    uint32_t keysym = xkb_state_key_get_one_sym (state, key);
+    uint32_t unicode = xkb_state_key_get_utf32 (state, key);
+
+    struct wpe_input_keyboard_event event;
```

Comment:
I would rather use C99 designated initializers here, as it guarantees that members not listed explicitly for initialization get filled with zeros:

```c
enum libinput_key_state key_state = libinput_event_keyboard_get_key_state (key_event);

struct wpe_input_keyboard_event event = {
    .time = libinput_event_keyboard_get_time (key_event),
    .key_code = keysym,
    .hardware_key_code = unicode,
    .pressed = (key_state == LIBINPUT_KEY_STATE_PRESSED),
};
```

This way we don't risk having member initialized with garbage in case the layout of the `struct` changes in the future.